aemcmc/gibbs.py

```diff
@@ -13,52 +18,143 @@
 )
 
 
-def update_beta_low_dimension(rng, omega, lambda2tau2_inv, X, z):
+def update_beta_low_dimension(
+    srng: RandomStream,
+    omega: TensorVariable,
+    lmbdatau_inv: TensorVariable,
+    X: TensorVariable,
+    z: TensorVariable,
+) -> TensorVariable:
     Q = X.T @ (omega[:, None] * X)
     indices = at.arange(Q.shape[1])
     Q = at.subtensor.set_subtensor(
         Q[indices, indices],
-        at.diag(Q) + lambda2tau2_inv,
+        at.diag(Q) + lmbdatau_inv,
     )
-    return multivariate_normal_rue2005(rng, X.T @ (omega * z), Q)
+    return multivariate_normal_rue2005(srng, X.T @ (omega * z), Q)
 
 
-def update_beta_high_dimension(rng, omega, lambda2tau2_inv, X, z):
-    return multivariate_normal_cong2017(rng, lambda2tau2_inv, omega, X, z)
+def update_beta_high_dimension(
+    srng: RandomStream,
+    omega: TensorVariable,
+    lmbdatau_inv: TensorVariable,
+    X: TensorVariable,
+    z: TensorVariable,
+) -> TensorVariable:
+    return multivariate_normal_cong2017(srng, lmbdatau_inv, omega, X, z)
 
 
-def update_beta(rng, omega, lambda2tau2_inv, X, z):
+def update_beta(
+    srng: RandomStream,
+    omega: TensorVariable,
+    lmbdatau_inv: TensorVariable,
+    X: TensorVariable,
+    z: TensorVariable,
+) -> TensorVariable:
     return ifelse(
         X.shape[1] > X.shape[0],
-        update_beta_high_dimension(rng, omega, lambda2tau2_inv, X, z),
-        update_beta_low_dimension(rng, omega, lambda2tau2_inv, X, z),
+        update_beta_high_dimension(srng, omega, lmbdatau_inv, X, z),
+        update_beta_low_dimension(srng, omega, lmbdatau_inv, X, z),
```

Comment:
Not related to this PR, but we should make sure that these `ifelse`s get rewritten to their respective branches when the conditions are constant/static graphs.

In other words, it would be unfortunate to evaluate those conditions unnecessarily on each iteration.